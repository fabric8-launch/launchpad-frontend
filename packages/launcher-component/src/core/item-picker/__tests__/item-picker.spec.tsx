import * as React from 'react';
import { cleanup, render, fireEvent } from 'react-testing-library';
import ItemPicker from '../item-picker';
import { mockItems } from './mock-items';

afterEach(cleanup);

describe('<ItemPicker />', () => {
  it('renders the ItemPicker correctly', () => {
    const comp = render(<ItemPicker group="test" value="value" onChange={() => {}} items={mockItems} />);
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it('pick one of the items correctly', () => {
    let value: string = '';
    const change = (picked: string) => value = picked;
    const comp = render(<ItemPicker group="test" value="value" onChange={change} items={mockItems} />);
    const item1Radio = comp.getByLabelText('Choose item1 as test');
    fireEvent.click(item1Radio);
    expect(value).toBe('item1');
  });
});
