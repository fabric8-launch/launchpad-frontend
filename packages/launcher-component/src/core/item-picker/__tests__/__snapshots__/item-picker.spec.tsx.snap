// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemPicker /> renders the ItemPicker correctly 1`] = `
<DocumentFragment>
  <div
    class="pf-l-gallery pf-m-gutter"
  >
    <div>
      <article
        class="pf-c-card card"
      >
        <div
          class="pf-c-card__header header"
        >
          <div
            class="pf-c-radio"
          >
            <input
              aria-invalid="false"
              aria-label="Choose item1 as test"
              class="pf-c-radio__input"
              id="radio-choose-item1-as-test"
              name="test"
              type="radio"
              value="item1"
            />
          </div>
          <h1
            class="pf-c-title pf-m-lg"
          >
            The test item1
          </h1>
        </div>
        <div
          class="pf-c-card__body body"
        >
          <img />
          <p />
        </div>
      </article>
    </div>
    <div>
      <article
        class="pf-c-card card"
      >
        <div
          class="pf-c-card__header header"
        >
          <div
            class="pf-c-radio"
          >
            <input
              aria-invalid="false"
              aria-label="Choose item2 as test"
              class="pf-c-radio__input"
              id="radio-choose-item2-as-test"
              name="test"
              type="radio"
              value="item2"
            />
          </div>
          <h1
            class="pf-c-title pf-m-lg"
          >
            The test item2
          </h1>
        </div>
        <div
          class="pf-c-card__body body"
        >
          <img />
          <p />
        </div>
      </article>
    </div>
  </div>
</DocumentFragment>
`;
